### DISCUSSION TITLE
Replace Makefile with meson.build

### DISCUSSION THREAD

**Discussion by s1341** [wazuh/wazuh#15821] [Ideas]
Hi,

Would you guys be open to replacing the current (imo ungainly) Makefile with a meson build?

I have done the heavy lifting of porting the make to meson already, and just need to add support for unit-testing and different agent/server targets to conditionally build those parts which need it. 

The benefits of the meson build is that we can detect (using pkg-config or cmake) the dependencies we need, so it is not necessary to vendor and build them ourselves.

You can see my wip meson.build here: https://github.com/s1341/wazuh/blob/meson_build/src/meson.build

(The reason that I embarked on this adventure, is because I am trying to package wazuh for nixos, and I needed a way to use existing dependencies. Patching the existing M\akefile quickly became very unwieldy.)

Interested in comments and feedback.
s1341

**Comment by MiguelazoDS** [comment]
Hi @s1341, 

Thanks for the proposal!, now we have a PR to be reviewed that changes the CMakeFiles configuration that manages the C++ code in our project, avoiding repeated statements and improving the readability to make it easier to extend later on (https://github.com/wazuh/wazuh/pull/15727).
This is not exactly what you're requesting, but this change makes us think about the unmanageable Makefile that we currently have. The main idea was to move everything to Cmake and then use Ninja build. I do not have too much knowledge about Meson, but if I'm not mistaken, it also generates a Ninja build. 
This is not a trivial task since we have to compile our project on multiple platforms, and in some of them may be complicated to make changes.
We will consider your comment when we address this, but I'm afraid this is not a high priority now. 

Thanks again. Regards!
